cs/test/compat/Program.cs

```diff
@@ -106,6 +106,11 @@ static void Write<R>(R reader, OutputStream output, string toProtocol)
                 var writer = new CompactBinaryWriter<OutputStream>(output);
                 Serialize.To(writer, Deserialize<Compat>.From(reader));
             }
+            else if (toProtocol == "compact2")
+            {
+                var writer = new CompactBinaryWriter<OutputStream>(output, 2);
+                Serialize.To(writer, Deserialize<Compat>.From(reader));
+            }
```

Comment:
I think this is not used unless you enable it [here](https://github.com/Microsoft/bond/blob/eb19aef894cf9c884654c14fdae0b65fdc1c0aea/cpp/test/compat/compat.cmake#L18-L21).